### ISSUE TITLE
gx: prefix does not appear on animationupdate kml

### ISSUE THREAD

**Issue by joeholdsworth** [samcragg/sharpkml#31]
Firstly thank you for this excellent library.

I am using the Google extended (GX) kml features.

When I look at the saved kml the gx tag only appears on the tour section. I.e. 
`<gx:tour>.....</gx:tour>`

I'm expecting it to appear on the `<animationupdate>` tag like this, `<gx:animationupdate>` but it doesn't. 

When I load the kml into Google earth the animation won't work unless I manually replace the `<animationupdate>` sections with `<gx:animationupdate>`.

What am I doing wrong or is this a bug?

**Comment by samcragg** [comment]
Thanks for the kind words and I'm sorry to hear that it's not working as intended. I've been able to reproduce the issue and, unfortunately, the library has set the correct namespace (if you notice the `Playlist` and `Update` elements switches back to the KML namespace):
```xml
<?xml version="1.0" encoding="utf-8"?>
<kml xmlns:gx="http://www.google.com/kml/ext/2.2" xmlns="http://www.opengis.net/kml/2.2">
  <Document>
    <name>gx:AnimatedUpdate example</name>
    <Placemark id="mountainpin1">
        ...
    </Placemark>
    <gx:Tour>
      <name>Play me!</name>
      <Playlist xmlns="http://www.google.com/kml/ext/2.2">
        <FlyTo>
          ...
        </FlyTo>
        <AnimatedUpdate>
          <duration>5</duration>
          <Update xmlns="http://www.opengis.net/kml/2.2">
            ...
          </Update>
        </AnimatedUpdate>
      </Playlist>
    </gx:Tour>
  </Document>
</kml>
```
So because we use the XML writer from the framework, the generated KML is technically correct and Google Earth doesn't handle the KML namespaces correctly (which is a known thing).

Having said all that, let me look at what I can to change it to make it generate the elements with the expected prefixes, as I agree with you, it should just work and it's not at the moment.

I'll update this issue later with more information when I have some more information.

**Comment by samcragg** [comment]
Actually, ignore the above, it's wrong. The culprit is I'm not using the `XmlNamespaceManager` correctly - I can't remember now why it's being used but looks like that's the source of where it's getting the wrong prefixes from.

Let me fix it and publish a new version of the library. I'm also going to add an option to automatically add the namespaces and their prefix to the root element too, with the option of turning it off if performance suffers too much.

**Comment by joeholdsworth** [comment]
Hi Sam, How's the update going, thanks for fixing this..

**Comment by samcragg** [comment]
I've just published a new version, 5.1.3, that fixes the issue; it should be available on NuGet shortly (it's been uploaded, but can take up to an hour to validate it and become available).

Thanks for reporting the issue and let me know of any other problems.
